azurerm/internal/services/storage/storage_account_resource_test.go

```diff
@@ -36,864 +39,694 @@ func TestValidateArmStorageAccountName(t *testing.T) {
+			),
 		},
+		data.ImportStep(),
 	})
 }
 
-func TestAccAzureRMStorageAccount_writeLock(t *testing.T) {
+func TestAccStorageAccount_writeLock(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_storage_account", "test")
+	r := StorageAccountResource{}
 
-	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { acceptance.PreCheck(t) },
-		Providers:    acceptance.SupportedProviders,
-		CheckDestroy: testCheckAzureRMStorageAccountDestroy,
-		Steps: []resource.TestStep{
-			{
-				Config: testAccAzureRMStorageAccount_writeLock(data),
-			},
-			{
-				// works around a bug in the test suite where the Storage Account won't be re-read after the Lock's provisioned
-				Config: testAccAzureRMStorageAccount_writeLock(data),
-				Check: resource.ComposeTestCheckFunc(
-					testCheckAzureRMStorageAccountExists(data.ResourceName),
-					resource.TestCheckResourceAttr(data.ResourceName, "account_tier", "Standard"),
-					resource.TestCheckResourceAttr(data.ResourceName, "account_replication_type", "LRS"),
-					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "1"),
-					resource.TestCheckResourceAttr(data.ResourceName, "tags.environment", "production"),
-					resource.TestCheckResourceAttr(data.ResourceName, "primary_connection_string", ""),
-					resource.TestCheckResourceAttr(data.ResourceName, "secondary_connection_string", ""),
-					resource.TestCheckResourceAttr(data.ResourceName, "primary_blob_connection_string", ""),
-					resource.TestCheckResourceAttr(data.ResourceName, "secondary_blob_connection_string", ""),
-					resource.TestCheckResourceAttr(data.ResourceName, "primary_access_key", ""),
-					resource.TestCheckResourceAttr(data.ResourceName, "secondary_access_key", ""),
-				),
-			},
-			data.ImportStep(),
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.writeLock(data),
 		},
+		{
+			// works around a bug in the test suite where the Storage Account won't be re-read after the Lock's provisioned
+			Config: r.writeLock(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("account_tier").HasValue("Standard"),
+				check.That(data.ResourceName).Key("account_replication_type").HasValue("LRS"),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("tags.environment").HasValue("production"),
+				resource.TestCheckResourceAttr(data.ResourceName, "primary_connection_string", ""),
+				resource.TestCheckResourceAttr(data.ResourceName, "secondary_connection_string", ""),
+				resource.TestCheckResourceAttr(data.ResourceName, "primary_blob_connection_string", ""),
+				resource.TestCheckResourceAttr(data.ResourceName, "secondary_blob_connection_string", ""),
+				resource.TestCheckResourceAttr(data.ResourceName, "primary_access_key", ""),
+				resource.TestCheckResourceAttr(data.ResourceName, "secondary_access_key", ""),
```

Comment:
COuld we use the check.That functions here?

Reply:
Oops, I missed this one